File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var dict = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; ++i) {
            if (dict.TryGetValue(target - nums[i], out int index)) 
                return new int[] { index, i };
            dict[nums[i]] = i;
        }
        return new int[] { -1, -1 };
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        int n = nums.size();
        for (int i=0; i<n; ++i) {
            int rem = target - nums[i];
            if (umap.find(rem) != umap.end())
                return {umap[rem], i};
            umap[nums[i]] = i;
        }
        return {-1, -1};
    }
};